### PR TITLE
Add GitHub Action to randomly assign a PR reviewer

### PR DESCRIPTION
**Pull request by ucey-star**
## Description

This PR introduces a GitHub Action to randomly assign a reviewer when a pull request (PR) is created. The action leverages a custom script to select a random reviewer from a predefined list of GitHub usernames and automatically assigns that user to the PR.

The reason for implementing this feature is to ensure an even distribution of PR review responsibilities within the team, encouraging collaboration and preventing bottlenecks where the same reviewer is overloaded. Additionally, the use of this automation reduces the need for manual intervention in the PR review assignment process.

Changes include:

- A GitHub Action file (assign-reviewer.yml) located in .github/workflows/, which triggers when a new pull request is opened.
- A script to randomly select a reviewer from an array of team members.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (adding or updating tests)
- [ ] Documentation update

## Testing Instructions

To test this:
- After you make a PR, check if it got randomly assigned to a member of the team

## Screenshots (if applicable)

If your changes include visual components, add screenshots showing the affected pages or elements.

### CODE REVIEW

File: .history/.github/workflows/assign-reviewer_20241021120127.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Assign Random Reviewer
```
Comment (lewiskyron) [inline .history/.github/workflows/assign-reviewer_20241021120127.yml:1]: I am not sure if we want to add the .history folder to the codebase. Did you push this or was this generated by github. These seem to be some cache files that do not need to be merged into the codebase.

Comment (lewiskyron) [reply .history/.github/workflows/assign-reviewer_20241021120127.yml:1]: Also please add instructions on how we can test that this works in the PR Testing section.

Comment (ucey-star) [reply .history/.github/workflows/assign-reviewer_20241021120127.yml:1]: > I am not sure if we want to add the .history folder to the codebase. Did you push this or was this generated by github. These seem to be some cache files that do not need to be merged into the codebase.

I added the /history to .gitignore

Comment (ucey-star) [reply .history/.github/workflows/assign-reviewer_20241021120127.yml:1]: > Also please add instructions on how we can test that this works in the PR Testing section.

done

Comment (ucey-star) [reply .history/.github/workflows/assign-reviewer_20241021120127.yml:1]: > I am not sure if we want to add the .history folder to the codebase. Did you push this or was this generated by github. These seem to be some cache files that do not need to be merged into the codebase.

oops. Actually I just saw the problem with this. My bad!

**Pull request closed without merging**